lib/utils/parseOptions.js

```diff
@@ -0,0 +1,77 @@
+import yargs from 'yargs';
+import {EOL} from 'os';
+
+import sanitizer from './sanitizer';
+import {CONSTANTS, ERRORS} from './constants';
+
+function parseSearchString (args) {
+  const searchErrors = [];
+  let searchString = args[0];
+
+  if (searchString && typeof(searchString) === 'string') {
+    searchString = sanitizer(searchString);
+  } else {
+    searchErrors.push(ERRORS.NO_SEARCH_STRING);
+  }
+
+  return {
+    searchString,
+    searchErrors
+  }
+}
+
+function parseFileExtensions (argv) {
+  const extensionErrors = [];
+  let fileExtensions = CONSTANTS.FILE_EXTENSION_DEFAULT;
+  let extensionError    = false;
+
+  const passedExtensions = argv.f;
+  if (passedExtensions) {
+    if (typeof(passedExtensions) === 'string') {
+      fileExtensions = passedExtensions.replace(/[\.\s]/g, '')
+                                       .split(',')
+                                       .filter(extension => !!extension); // filter empty strings
+      switch(fileExtensions.length) {
+        case 0:
+          extensionError = true;
+          break;
+        case 1:
+          fileExtensions = fileExtensions[0];
+          break;
+        default:
+          fileExtensions = '{' + fileExtensions.join(',') + '}';
+      }
+    } else {
+      extensionError = true;
+    }
+  }
+
+  if (extensionError) {
+    extensionErrors.push(ERRORS.NO_FILE_EXTENSION);
+  }
+
+  return {
+    fileExtensions,
+    extensionErrors
+  }
+}
+
+module.exports = args => {
+  const argv = yargs.usage(CONSTANTS.COMMAND_USAGE)
+                    .example(CONSTANTS.COMMAND_EXAMPLE)
+                    .alias('f', 'file')
+                    .parse(args);
+  const optionErrors = [];
+
+  const {searchString, searchErrors} = parseSearchString(argv._);
+  optionErrors.push(...searchErrors);
+
+  const {fileExtensions, extensionErrors} = parseFileExtensions(argv);
+  optionErrors.push(...extensionErrors);
+
+  return {
+    searchString,
+    fileExtensions,
+    optionErrors
+  };
+}
```

Comment:
Also please be mindful of how other files have been written. You should be following the same patterns.
- `export default clone;` at the bottom of files https://github.com/hmrc/component-finder/blob/master/lib/clone.js#L70
- const function expressions `const clone = (services, task = cloneTask) => {` https://github.com/hmrc/component-finder/blob/master/lib/clone.js#L49

> All code in any code-base should look like a single person typed it, no matter how many people contributed. 
- [Rick Waldron](https://github.com/rwaldron/idiomatic.js/#all-code-in-any-code-base-should-look-like-a-single-person-typed-it-no-matter-how-many-people-contributed)